wordpress-mobile/WordPress-iOS
# Feature: Add and set a property to mark local authors as deleted.

This is one PR in a series towards implementing #9015.

**Current behavior:** Authors who are deleted from a blog continue to accumulate in the local store of authors in the app.

**What this change does:** Authors who are deleted from a blog continue to accumulate in the local store of authors in the app, but have the property `deletedFromBlog` set to `true`. All others are set to `false`.

**Why this change was needed:** This will prevent users from seeing deleted authors in the list to choose from once #9015 is implemented.

### To test:

1. Prepare a WordPress.com or self-hosted WordPress blog with multiple users.
2. Place a breakpoint on `BlogService+BlogAuthors.swift:36` and load the "My Site" view for the blog.
3. When the breakpoint is caught, observe the output of: `po remoteUsers` and `po blog.authors`.
- The `remoteUsers` array should contain all of the blog's remote users.
- The `blog.authors` set should match number of remote users, and the `deletedFromBlog` property should be `0` for each.
4. Continue execution of the app.
5. Delete one of the users from the blog using the WordPress Dashboard.
6. Restart WPiOS or choose "Blog Posts" and press "Back" to return to the "My Site" view.
7. When the breakpoint is caught, observe the output of the commands from step 3.
- The `remoteUsers` array should contain one less remote user (the user that was deleted is no longer in that array).
- The `blog.authors` set should still match the number of remote users from step 3, but the `deletedFromBlog` property should equal `1` for the deleted author.

## Regression Notes
### 1. Potential unintended areas of impact

- None identified.

### 2. What I did to test those areas of impact (or what existing automated tests I relied on)

**Metadata for a deleted author (Display Name, Avatar icon) should still appear when viewing the History of a Post.**

1. Make changes to a Post as multiple users (e.g. User A and User B have both made updates to the same Post).
2. With the Post presented in the app, tap the horizontal ellipses (More Options) icon at the top right.
3. Select "History".
4. Observe the display names and avatars next to each revision.
5. Remove an author from the blog.
6. Restart the app.
7. View the history for the same Post.
8. There should be no changes or regressions.

### 3. What automated tests I added (or what prevented me from doing so)

- Added `BlogServiceAuthorTests`.

PR submission checklist:

- [x] I have documented the migration.
